src/edu/psu/sweng500/team8/gui/SudokuGUI.java

```diff
@@ -346,12 +382,162 @@ private void radHardActionPerformed(java.awt.event.ActionEvent evt) {// GEN-FIRS
 	private void jButton11ActionPerformed(java.awt.event.ActionEvent evt) {// GEN-FIRST:event_jButton11ActionPerformed
 		// TODO add your handling code here:
 		// Create a file chooser
+		
 		final JFileChooser fc = new JFileChooser();
-		// In response to a button click:
+		fc.setFileFilter(new FileFilter()
+        {
+           @Override
+           public boolean accept(File file)
+           {
+              return (file.getName().toUpperCase().endsWith(".SUDOKU") || file.isDirectory());
+           }
+
+           @Override
+           public String getDescription()
+           {
+              return "Sudoku files";
+           }
+        });
+		
+		fc.setAcceptAllFileFilterUsed(false);
 		int returnVal = fc.showOpenDialog(getComponent(0));
 
+		if(returnVal == JFileChooser.APPROVE_OPTION){
+		
+			SavePackage savePackage = null;
+	      try
+	      {
+	         FileInputStream fileIn = new FileInputStream(fc.getSelectedFile().getAbsolutePath());
+	         ObjectInputStream in = new ObjectInputStream(fileIn);
+	         savePackage = (SavePackage) in.readObject();
+	         in.close();
+	         fileIn.close();
+	         this.loadSession(savePackage);
+	      }catch(IOException i)
+	      {
```

Comment:
Inconsistent brace style -- should be on the end of the preceding line. Same below.